File: android/app/src/main/java/com/vegettable/app/model/Recipe.java
package com.vegettable.app.model;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class Recipe {
    @SerializedName("name")
    private String name;

    @SerializedName("description")
    private String description;

    @SerializedName("ingredients")
    private List<String> ingredients;

    @SerializedName("difficulty")
    private String difficulty; // "easy" | "medium" | "hard"

    @SerializedName("cookTimeMinutes")
    private int cookTimeMinutes;

    public String getName() { return name; }
    public String getDescription() { return description; }
    public List<String> getIngredients() { return ingredients; }
    public String getDifficulty() { return difficulty; }
    public int getCookTimeMinutes() { return cookTimeMinutes; }
}

File: android/app/src/main/java/com/vegettable/app/model/Recipe.kt
package com.vegettable.app.model

import com.google.gson.annotations.SerializedName

class Recipe {
    @JvmField
    @SerializedName("name")
    val name: String? = null

    @JvmField
    @SerializedName("description")
    val description: String? = null

    @SerializedName("ingredients")
    val ingredients: MutableList<String?>? = null

    @SerializedName("difficulty")
    val difficulty: String? = null // "easy" | "medium" | "hard"

    @JvmField
    @SerializedName("cookTimeMinutes")
    val cookTimeMinutes: Int = 0
}
